tests/torch/test_llama_decoder.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+import pytest
+
+pytest.importorskip("torch", reason="PyTorch not installed. Please install with: pip install dace[ml]")
+import torch
+import torch.nn as nn
+from transformers.models.llama.modeling_llama import LlamaDecoderLayer, LlamaConfig
+from dace.frontend.python.module import DaceModule
+from tests.utils import torch_tensors_close
+
+
+# Create a wrapper module that handles the position embeddings internally
+class LlamaDecoderLayerWrapper(nn.Module):
+
+    def __init__(self, decoder_layer, config):
+        super().__init__()
+        self.decoder_layer = decoder_layer
+        self.config = config
+
+        # Create rotary embeddings as part of the wrapper
+        from transformers.models.llama.modeling_llama import LlamaRotaryEmbedding
+        self.rotary_emb = LlamaRotaryEmbedding(config)
+
+    def forward(self, hidden_states, attention_mask, position_ids):
+        # Generate position embeddings
+        cos, sin = self.rotary_emb(hidden_states, position_ids)
+
+        # Call the decoder layer
+        outputs = self.decoder_layer(
+            hidden_states=hidden_states,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            position_embeddings=(cos, sin),
+            past_key_value=None,
+            output_attentions=False,
+            use_cache=False,
+        )
+
+        # Return only the hidden states (first element of the tuple)
+        return outputs[0]
+
+
+@pytest.mark.torch
+def test_llama_decoder(sdfg_name: str):
```

Comment:
isn't this a copy of the test in autodiff/torch, but now forward pass only?
I think the duplicates can be removed from here to save on CI minutes.